core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java

```diff
@@ -317,12 +316,10 @@ private static Map<SchemaTableName, List<ColumnMetadata>> doListTableColumns(Ses
                         actualTableName = redirection.redirectedTableName().get();
                         targetTableHandle = redirection.tableHandle().orElseThrow();
                     }
-                    catch (TrinoException e) {
-                        // Ignore redirection errors
-                        if (e.getErrorCode().equals(TABLE_REDIRECTION_ERROR.toErrorCode())) {
-                            return;
-                        }
-                        throw e;
+                    catch (RuntimeException e) {
+                        // Ignore errors during listing
+                        MetadataManager.handleListingError(e, prefix);
+                        return;
```

Comment:
The old code was correct. If this is an issue with the new Glue metastore code, it should be retrying.  If we want to ignore some falures during listing, we should explicitly be handling those errors.  Otherwise listing is just a "show whatever happens to be online and it may or may not be correct".

Reply:
Remove this change as original issue needs only GlueHiveMetastore fix.